### PR TITLE
[close #117] Br dependence cleanup

### LINKED ISSUE #117: [TiKV-BR] Remove dependence to etcd, pd and tidb
### 1. Describe the bug
TiKV-BR depends on many tidb and pd packages, which cause some dependence problems. tidb/br has removed the dependence to PD. Please remove the dependence to PD in TiKV-BR and reduce dependence to TiDB also.

### 5. What is your migration tool and TiKV version? (Required)

- TiKV BR:

### PR DESCRIPTION
**Pull request by haojinming**
### What problem does this PR solve?

Issue Number: close #117 

Problem Description:  TiKV-BR shoudl not has dependece on `etcd`, `PD` and `TiDB` codes.

### What is changed and how does it work?

1. Remove dependece on `etcd` and `PD` and reduce rependence on `TiDB`, only depend on `PDClient` and `client-go`
2. Clean some useless codes, like table id encoding/decoding, which is not suitable for `TiKV-BR`.

Why remove dependence on `TiDB`:   
TiDB now use go1.18 and some code cannot be compiled with go1.16 .
However, there still some dependences which are not easy to remove, like `github.com/pingcap/tidb/util/encrypt`, keep this for future work.

Why remove dependence on `PD`:   
PD use low version `etcd` which does not support `grpc` with version higher than 1.29.0, but latest `client-go` use 1.43.0. If `TiKV-BR` want to use latest `client-go`, `TiKV-BR` need ungrade `etcd` to higher version with support of `grpc` 1.43.0 or remove the dependence. The latter seems better. Latest TiDB has achieved this, we reference the solution in `TiDB`, which defines `pdtypes` package in `TiDB` and use `pdtypes`  instead of `github.com/tikv/pd`. 

Why remove dependence on `etcd`:   
`TiKV-BR` hase no relation with `etcd`, should remove the dependence.

### Code changes

- Has exported function/method change
- Has exported variable/fields change
- Has interface methods change

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Side effects

- No side effects

### Related changes

- No related changes

### CODE REVIEW

File: br/pkg/restore/merge.go [hunk 1]
```diff
@@ -106,22 +104,8 @@ func MergeFileRanges(
 		if leftKeys+rightKeys > splitKeyCount {
 			return false
 		}
-		// Do not merge ranges in different tables.
```
Comment (pingyu) [inline br/pkg/restore/merge.go:109]: Maybe we should avoid merge regions from different kv mode or keyspace in the future.

File: br/pkg/restore/range.go [hunk 2]
```diff
@@ -28,36 +26,25 @@ func SortRanges(ranges []rtree.Range, rewriteRules *RewriteRules) ([]rtree.Range
 	rangeTree := rtree.NewRangeTree()
 	for _, rg := range ranges {
 		if rewriteRules != nil {
-			startID := tablecodec.DecodeTableID(rg.StartKey)
```
Comment (pingyu) [inline br/pkg/restore/range.go:31]: We don't need rewrite rules at all ?

**Review by pingyu — COMMENTED** [review]
Rest LGTM~

**Review by zeminzhou — DISMISSED** [review]

File: br/pkg/restore/range.go (hunk 2 above)
Comment (haojinming) [reply br/pkg/restore/range.go:31]: I'm not sure for now. Maybe need rewrite rule in the future. So keep to rewrite rule placement and just remove the table id encoding/decoding parts.

File: br/pkg/restore/merge.go (hunk 1 above)
Comment (haojinming) [reply br/pkg/restore/merge.go:109]: OK, will handle this cases when tikv-br support different keyspaces and modes.

File: br/cmd/br/cmd.go [hunk 3]
```diff
@@ -157,27 +133,6 @@ func Init(cmd *cobra.Command) (err error) {
 
 func startPProf(cmd *cobra.Command) error {
 	// Initialize the pprof server.
-	statusAddr, err := cmd.Flags().GetString(FlagStatusAddr)
```
[the remark this answers is not in the document]
Comment (haojinming) [reply br/cmd/br/cmd.go:160]: OK

**Review by pingyu — APPROVED** [review]
LGTM~

**Review by zeminzhou — APPROVED** [review]

**Pull request merged**